fix headers TrackPropagation/RungeKutta

mrodozov opened the pull request:
Fix headers related to #15248

Commit:
Update the ADC->fC table, simplify the charge reconstruction algorithm, enable the MET filter

Commit:
update with code-check results

Commit:
remove old commented code

Commit:
Fix parameter name and update threshold to 5000

Commit:
update with code review comments

Commit:
clone hcalLaserEventFilter

boudoul committed:
Adding relevant xmls for 20% and 50% material variation, change RelVal sequences, update documentation

anmehta committed:
adding files for FSQ DQM code

anmehta committed:
scram-build-cross-checks done

anmehta committed:
flag to check if a collection is not found in the WF

Commit:
remove comments and update to range based loop

hatakeyamak committed:
Clean up the HGC validation folder structure

anmehta committed:
implemented suggestions from DLange

ianna committed:
Cleanup DD compact view

hatakeyamak committed:
style update patch

anmehta committed:
fixed ClangBuild

schneiml committed:
Adapt @express sequence

... so that it can actually be used for Express processing at T0.

schneiml committed:
Fix HLTMon typo.

tanmaymudholkar committed:
The cuts on rechits included in the ECAL timing plots were out of date. Especially in the forward region most of the rechits passing the cuts were pure noise and created a few false alarms. This commit introduces these changes:

(1) Update thresholds on rechit energy for rechits to be included in the timing plots.

(2) Introduce cut on timing error to match offline processing.

(3) Implement separate energy thresholds for forward region in EE.

anmehta committed:
implemented suggestion from @fabiocos

ianna committed:
Cleanup DD Compact View

Commit:
change to const reference

jfernan2 committed:
Removal of unused DQM Online plots and LS label fixing

jfernan2 committed:
remove extra plots

jfernan2 committed:
Last fix for empty folders

Commit:
fix headers in SUSYBSMAnalysis/HSCP

Commit:
fix headers in DataFormats/Common

Commit:
fix headers in ValidationPerformance

Commit:
fix headers in RecoTracker/FinalTrackSelectors

Commit:
fix headers in Fireworks/Core

jfernan2 committed:
Last fixes

Commit:
fix headers in PhysicsTools/PatExamples

Commit:
fix headers in EventFilter/EcalTBRawToDigi

Commit:
fix headers RecoEgamma/EgammaMCTools

Commit:
fix header in DataFormats/TrackerRecHit2D

mdhildreth committed:
missing commas...

Dr15Jones committed:
modernized DTHitQualityUtils

DTHitQualityUtils was a class with only static members. This was
changed to be a namespace instead.
Converted global 'debug' variable to atomic.

Dr15Jones committed:
Made anonymous namespace variable constexpr

The variable mirrorMinusWheels is only used in the file and is
only set as part of its declaration. This allows it to be constexpr
which also fixes the thread-safety warning.

mdhildreth committed:
add spurious new lines to make geometry test pass

Commit:
fix headers TrackPropagation/RungeKutta

mrodozov commented:
please test

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23326/4826

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
A new Pull Request was created by  @mrodozov (Mircho Rodozov) for master.

It involves the following packages:

TrackPropagation/RungeKutta


@perrotta, @cmsbuild, @slava77 can you please review it and eventually sign? Thanks.
@makortel, @felicepantaleo, @ebrondol, @VinInn this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cmsbuild committed:
Merge pull request #23203 from mdhildreth/Phase2TrackerGeometriesModMaterial

Phase2 tracker geometries: Add 20% and 50% material variation

Commit:
fix header in L1Trigger/L1TCaloLayer1

Commit:
Utilities/StaticAnalyzers

Commit:
fix header in RecoPixelVertexing/PixelTriplets

cmsbuild commented:
+1
Tested at: 328080eeba3882fd5513d2f340ca5244e2dca142
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23326/28181/summary.html

cmsbuild commented:
Comparison job queued.

Commit:
Update L1TCaloLayer1FetchLUTs.hh

Include also vector and array

Commit:
Update FWGUIEventDataAdder.h

Substitute inclusion of header with fwd declaration.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-23326/28181/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 31
 - DQMHistoTests: Total histograms compared: 2901706
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2901515
 - DQMHistoTests: Total skipped: 190
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 30 files compared)
 - Checked 128 log files, 14 edm output root files, 31 DQM output files

smuzaffar committed:
cleanup: move RecoBTag/DeepFlavour/src/sorting_modules.cc to plugins

jfernan2 committed:
Fix for FEDIntegrity_EvF test

Dr15Jones committed:
FunctorTask can now use a mutable lambda

Use std::move on the functor object allows the use of a mutable
lambda containing a std::unique_ptr.

Dr15Jones committed:
Added log method which takes a lambda

The log method allows complex code which is only needed for logging to only be run if the logging level is appropriate. This is done by passing a lambda to the method where the lambda has the signature '(auto &)' and is passed an object which uses operator<<.

Dr15Jones committed:
Switched PathAnalyzer to use LogInfo::log

Now all the computation needed for the log message is only done if the log would be seen based on the logging level.

cmsbuild committed:
Merge pull request #23276 from schneiml/patch-3

Adapt @express sequence for T0

cmsbuild committed:
Merge pull request #23283 from tanmaymudholkar/TimingCuts_master

New timing cuts for ECAL (for CMSSW master)

cmsbuild committed:
Merge pull request #23312 from mrodozov/fix-headers-DF/Common-pkg

fix headers in DataFormats/Common

cmsbuild committed:
Merge pull request #23314 from mrodozov/fix-headers-Vldtn/Prfmns-pkg

fix headers in ValidationPerformance and add header guard

cmsbuild committed:
Merge pull request #23315 from mrodozov/fix-headers-RcTrk/FnlTrkSektr-pkg

fix headers in RecoTracker/FinalTrackSelectors

cmsbuild committed:
Merge pull request #23317 from mrodozov/fix-hdrs-Fireworks/Core

fix headers in Fireworks/Core

cmsbuild committed:
Merge pull request #23321 from mrodozov/fix-headers-EventFilter/EcalTBRawToDigi-pkg

fix headers in EventFilter/EcalTBRawToDigi

cmsbuild committed:
Merge pull request #23322 from mrodozov/fix-hdrs-RecoEgamma/EgammaMCTools-pkg

fix headers RecoEgamma/EgammaMCTools

cmsbuild committed:
Merge pull request #23323 from mrodozov/fix-hdr-DataFormats/TrackerRecHit2D-pkg

fix header in DataFormats/TrackerRecHit2D

cmsbuild committed:
Merge pull request #23327 from mrodozov/fix-hdr-L1Trigger/L1TCaloLayer1-pkg

fix header in L1Trigger/L1TCaloLayer1

cmsbuild committed:
Merge pull request #23328 from mrodozov/fix-hdr-Utilities/StaticAnalyzers-pkg

fix header in Utilities/StaticAnalyzers

perrotta commented on TrackPropagation/RungeKutta/src/CartesianStateAdaptor.h:
```diff
@@ -1,6 +1,9 @@
 #ifndef CartesianStateAdaptor_H
 #define CartesianStateAdaptor_H
 
+#include "RKSmallVector.h"
```
Better to include headers with the full path
(I see that this is already not always satisfied by the current code: @mrodozov do you want to propagate it, at least into the files touched by this pull request?)

makortel replied:
@perrotta I agree in general (i.e. for headers in `interface`), but why would it be better for headers in `src` (or `plugins`) that ought to be implementation details of the library (plugin) and not to be `#include`'d elsewhere?

perrotta commented on TrackPropagation/RungeKutta/src/RKCartesianDerivative.h:
```diff
@@ -1,7 +1,7 @@
 #define RKCartesianDerivative_H
 
-class CartesianState;
+#include "CartesianState.h"
```
Better to include headers with the full path

VinInn replied:
not for local header

slava77 replied:
I agree: local headers (not in interface/) should be included like this.
And whoever is including A/B/src/B.h from a package C/D is doing it wrong.

perrotta commented on TrackPropagation/RungeKutta/src/RKCartesianDistance.h:
```diff
@@ -2,6 +2,7 @@
 
 #include "FWCore/Utilities/interface/Visibility.h"
+#include "DataFormats/GeometryVector/interface/Basic3DVector.h"
 #include "RKDistance.h"
```
e.g.

perrotta commented on TrackPropagation/RungeKutta/src/RKCurvilinearDistance.h:
```diff
@@ -1,6 +1,7 @@
 
+#include "DataFormats/GeometryVector/interface/Basic3DVector.h"
 #include "FWCore/Utilities/interface/Visibility.h"
 #include "RKDistance.h"
```
e.g.

perrotta commented on TrackPropagation/RungeKutta/src/RKOne4OrderStep.h:
```diff
@@ -2,6 +2,7 @@
 
 #include "FWCore/Utilities/interface/Visibility.h"
+#include "TrackPropagation/RungeKutta/src/RKDistance.h"
 #include "RK4OneStepTempl.h"
```
e.g.

cmsbuild committed:
Merge pull request #23347 from Dr15Jones/lambdaForMessageLogging

Added method 'log' to MessageLoggers to handle complex messaging

cmsbuild committed:
Merge pull request #23346 from Dr15Jones/mutableTaskFunctor

FunctorTask can now use a mutable lambda

cmsbuild committed:
Merge pull request #22974 from jkunkle/LasermonReadinessUpdate

Update the ADC->fC table, simplify the charge reconstruction algorith…

cmsbuild committed:
Merge pull request #23158 from anmehta/from-CMSSW_10_2_X_2018-05-06-2300_FSQDQMCode

adding files for FSQ DQM code

cmsbuild committed:
Merge pull request #23285 from ianna/dd-cleanup-dd-compact-view

DD Cleanup: DD Compact View

cmsbuild committed:
Merge pull request #23306 from mrodozov/fix-headers-SUSYBSMAnalysis/HSCP-pnk

fix headers in SUSYBSMAnalysis/HSCP

cmsbuild committed:
Merge pull request #23335 from smuzaffar/cleanup-RecoBTag-DeepFlavour

cleanup: move RecoBTag/DeepFlavour/src/sorting_modules.cc to plugins

cmsbuild committed:
Merge pull request #23318 from mrodozov/fix-headers-PhysicsTools/PatExamples-pkg

fix headers in PhysicsTools/PatExamples

alberto-sanchez committed:
addressing issue 23334, moving files to satisfy header-consistency checks

alberto-sanchez committed:
addressing issue 23334, moving files to satisfy header-consistency checks

alberto-sanchez committed:
addressing issue 23334, replacing headers

alberto-sanchez committed:
updating cfg for test and clean up snip

alberto-sanchez committed:
adding unit tests

alberto-sanchez committed:
applying code checks

perrotta commented:
Thank you @makortel and @VinInn 

Indeed, at the end my concern was probably more aesthetical than practical (since sometime the full path was written, e.g. in https://github.com/mrodozov/cmssw/blob/328080eeba3882fd5513d2f340ca5244e2dca142/TrackPropagation/RungeKutta/src/RKCartesianDistance.h#L5, and sometime it is not)

I will not insist on it, unless @mrodozov wants to profit and make uniform the includes in the files touched by this PR

cmsbuild committed:
Merge pull request #23348 from alberto-sanchez/issue-23334

Addressing Issue 23334

cmsbuild committed:
Merge pull request #23256 from hatakeyamak/Clean_HGC_relval_folders

Clean up the HGC validation folder structure

cmsbuild committed:
Merge pull request #23330 from mrodozov/fix-hdr-RecoPixelVertexing/PixelTriplets-pkg

fix header in RecoPixelVertexing/PixelTriplets

cmsbuild committed:
Merge pull request #23324 from Dr15Jones/modernizeDTHitQualityUtils

modernized Validation/DTRecHits

cmsbuild committed:
Merge pull request #23298 from jfernan2/RemoveDToldFolders102X

Removal of unused DT DQM Online plots and LS label fixing

Commit:
Merged refs/pull/23326/head from repository cms-sw with cms-merge-topic

Commit:
Include headers with full path for consistency

cmsbuild commented:
The code-checks are being triggered in jenkins.

mrodozov commented:
oops I messed it, I'll close it and propose another one with the changes for the headers

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-23326/4858

cmsbuild commented:
Pull request #23326 was updated. @ianna, @kpedro88, @fabozzi, @nsmith-, @rekovic, @thomreis, @alberto-sanchez, @vanbesien, @arizzi, @perrotta, @civanch, @monttj, @cmsbuild, @GurpreetSinghChahal, @davidlange6, @smuzaffar, @Dr15Jones, @efeyazgan, @mdhildreth, @jfernan2, @slava77, @fabiocos, @prebello, @vazzolini, @kmaeshima, @perrozzi, @dmitrijus, @alja, @franzoni, @gpetruc can you please check and sign again.

perrotta commented:
@mrodozov : I think the conclusion of the discussion  (that I started) was to NOT include the full path.
Therefore, the only one to be fixed is the only one where you actually already did it:
/blob/328080eeba3882fd5513d2f340ca5244e2dca142/TrackPropagation/RungeKutta/src/RKOne4OrderStep.h#L5
